### ISSUE TITLE
docs update: keycloak.md shows an old keycloak version and should be updated

### ISSUE THREAD

**Issue by mmueller-a9s** [argoproj/argo-cd#12495]
# Summary

The operator-manual to configure SSO with Keycloak seems to use a Keycloak version of ~13. By now, Keycloak version 20 is out and the UI has changed significantly.

# Motivation

If one wants to configure ArgoCD to use Keycloak for SSO, one wants to use this documentation to configure the client.

# Proposal

Go through the process and make new screenshots on how to configure the client properly.

# Additional Info
There is already an updated version Draft in my fork: [mmueller-a9s/argo-cd](https://github.com/mmueller-a9s/argo-cd/tree/docs_update_keycloak)

**Comment by mmueller-a9s** [comment]
@todaywasawesome I've created a Pull Request: https://github.com/argoproj/argo-cd/pull/12505
The functionality is the same between the versions. The only difference is that Keycloak removed the `/auth` from the path, but this has already been updated in https://github.com/argoproj/argo-cd/commit/e5e3b0f5790c167f82d9d306bdae768d69724434

I've configured a keycloak client in Keycloak v13 using the old documentation and updated Keycloak to v20. When I updated this documentation, I paid attention that the configuration is equal.
I also created a new keycloak client in Keycloak v20 using the new documentation and it worked.
